lib/cache.gi

```diff
@@ -34,6 +34,9 @@ function(func, defaults...)
 
     return function(val)
         local v, boundcpy;
+        if not IsPosInt(val) then
+            Error("<val> must be a positive integer");
+        fi;
```

Comment:
We could also pass in an (optional) "fallback" function, which is called if `val` is not a pos int. That would e.g. simplify writing `Bernoulli`, as the fallback function could deal with `n=0`, and also show a custom error message. But I wasn't sure how useful that is in general, so instead of risking to over-engineer, I figured we can start with a simple error message here, and move on if we e.g. find lots of things that need to memoize non-negative integers.